From 45018491270614729d18270a7c101b37ea2c867f Mon Sep 17 00:00:00 2001
From: Eric Sauvageau <merlin@asuswrt-merlin.net>
Date: Mon, 3 Aug 2026 23:29:02 -0400
Subject: [PATCH] libovpn: generate static key if TLS control channel security
 is set to tls-auth and we have a non-V1 static key present (like a V2 key)

(cherry picked from commit 817605edb14b6fd5a37e27e03fffd5f5f4b74d04)
---
 release/src/router/libovpn/openvpn_setup.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/release/src/router/libovpn/openvpn_setup.c b/release/src/router/libovpn/openvpn_setup.c
index 6082bf68dd..0e1c217a33 100644
--- a/release/src/router/libovpn/openvpn_setup.c
+++ b/release/src/router/libovpn/openvpn_setup.c
@@ -1321,6 +1321,8 @@ void ovpn_write_static_key(ovpn_sconf_t *sconf, int unit) {
 			generate = 1;
 		if (sconf->tlscrypt == OVPN_TLS_CRYPT_V1 && !strcasestr(buffer, OVPN_PEM_HEADER_CRYPT_V1))
 			generate = 1;
+		if ((sconf->direction == 0 || sconf->direction == 1 || sconf->direction == 2) && !strcasestr(buffer, OVPN_PEM_HEADER_CRYPT_V1))
+			generate = 1;
 	} else {
 		generate = 1;
 	}
-- 
2.25.1

